### PR TITLE
03-2167:Support associating concept representation with questions

### PR DESCRIPTION
**Pull request by gitcliff**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
PR Adds the concept object  within the data structure (possible in a new "meta" property, so that we dont need to make subsequent backend calls to get more information about the concept associated with the question
## Screenshots
Screenshot 2024-04-16 at 10 40 17 PM

### CODE REVIEW

File: src/components/encounter/ohri-encounter-form.component.tsx [hunk 1]
```diff
@@ -260,9 +260,12 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
...
           //   2) use the concept display as the label if no label specified
-          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts);
-          field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
-          field.label = field.label ? field.label : matchingConcept?.display;
+          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts); 
+          field.meta = matchingConcept;
+          field.questionOptions.concept = field.meta ? field.meta?.uuid : field.questionOptions.concept;
+         
+
+          field.label = field.label ? field.label : field.meta?.display;
```
Comment (pirupius) [inline src/components/encounter/ohri-encounter-form.component.tsx:268]:
```suggestion
          field.meta = matchingConcept;
          field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
          field.label = field.label ? field.label : field.meta?.display;
```

File: src/components/inputs/content-switcher/ohri-content-switcher.component.tsx [hunk 2]
```diff
@@ -42,9 +42,10 @@ export const OHRIContentSwitcher: React.FC<OHRIFormFieldProps> = ({ question, on
   );
 
   useEffect(() => {
-    getConceptNameAndUUID(question.questionOptions.concept).then((conceptTooltip) => {
-      setConceptName(conceptTooltip);
-    });
+
```
Comment (pirupius) [inline src/components/inputs/content-switcher/ohri-content-switcher.component.tsx:45]: Remove empty line

File: src/components/inputs/file/file.component.tsx [hunk 3]
```diff
@@ -81,9 +81,9 @@ const File: React.FC<FileProps> = ({ question, onChange, handler }) => {
   };
 
   useEffect(() => {
-    getConceptNameAndUUID(question.questionOptions.concept).then((conceptTooltip) => {
-      setConceptName(conceptTooltip);
-    });
+    if (question.meta) {
+      setConceptName(question.meta?.display);
+   }
```
Comment (pirupius) [inline src/components/inputs/file/file.component.tsx:86]: This is starting to look repetitive. Is there an option to get rid of the useEffect all together and just infer the question.meta?.display or is it tied to the question being fully loaded within the useEffect?

File: src/utils/ohri-form-helper.ts [hunk 4]
```diff
@@ -45,11 +45,11 @@ export function inferInitialValueFromDefaultFieldValue(
...
-export function getConceptNameAndUUID(conceptUuid: string) {
-  return fetchConceptNameByUuid(conceptUuid).then((conceptName) => {
-    return `Concept Name: ${conceptName} \n UUID: ${conceptUuid}`;
-  });
-}
+// export function getConceptNameAndUUID(conceptUuid: string) {
+//   return fetchConceptNameByUuid(conceptUuid).then((conceptName) => {
+//     return `Concept Name: ${conceptName} \n UUID: ${conceptUuid}`;
+//   });
+// }
```
Comment (pirupius) [inline src/utils/ohri-form-helper.ts:52]: Delete if not used anymore

File: src/components/encounter/ohri-encounter-form.component.tsx (hunk 1 above)
Comment (samuelmale) [inline src/components/encounter/ohri-encounter-form.component.tsx:268]:
```suggestion
          const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts);
          field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
          field.label = field.label ? field.label : matchingConcept?.display;
          field.meta.concept = matchingConcept;
```

Take note of the concept path.

File: src/components/inputs/date/ohri-date.component.tsx [hunk 5]
```diff
@@ -129,9 +129,9 @@ const OHRIDate: React.FC<OHRIFormFieldProps> = ({ question, onChange, handler, p
   }, [encounterContext?.previousEncounter]);
 
   useEffect(() => {
-    getConceptNameAndUUID(question.questionOptions.concept).then((conceptTooltip) => {
-      setConceptName(conceptTooltip);
-    });
+    if (question.meta) {
+      setConceptName(question.meta?.display);
+    }
```
Comment (samuelmale) [inline src/components/inputs/date/ohri-date.component.tsx:134]: @gitcliff do we still need the encompassing `useEffect`?

File: src/components/inputs/file/file.component.tsx (hunk 3 above)
Comment (samuelmale) [reply src/components/inputs/file/file.component.tsx:86]: +100

**Review by samuelmale — CHANGES_REQUESTED** [review]
@gitcliff can you also remove functions like: `getConceptNameAndUUID` and probably `fetchConceptNameByUuid`?

File: src/components/inputs/date/ohri-date.component.tsx (hunk 5 above)
Comment (gitcliff) [reply src/components/inputs/date/ohri-date.component.tsx]: oh i see 
i have removed it @samuelmale

File: src/components/inputs/file/file.component.tsx (hunk 3 above)
Comment (gitcliff) [reply src/components/inputs/file/file.component.tsx]: thanks @pirupius i have removed the useEffect

File: src/components/inputs/location/ohri-encounter-location.component.tsx [hunk 6]
```diff
@@ -42,18 +45,12 @@ export const OHRIEncounterLocationPicker: React.FC<{ question: OHRIFormField; on
...
-    });
-  }, [conceptName]);
-
   return encounterContext.sessionMode == 'view' || encounterContext.sessionMode == 'embedded-view' ? (
     <div className={styles.formField}>
       <OHRIFieldValueView
         label={question.label}
         value={field.value ? field.value.display : field.value}
-        conceptName={conceptName}
+        conceptName={conceptNameDisplay}
```
Comment (samuelmale) [inline src/components/inputs/location/ohri-encounter-location.component.tsx:53]: @gitcliff why not:

```suggestion
        conceptName={question.meta?.concept?.display;}
```

File: src/components/encounter/ohri-encounter-form.component.tsx [hunk 7]
```diff
@@ -263,6 +263,8 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
           const matchingConcept = findConceptByReference(field.questionOptions.concept, concepts);
           field.questionOptions.concept = matchingConcept ? matchingConcept.uuid : field.questionOptions.concept;
           field.label = field.label ? field.label : matchingConcept?.display;
+          field.meta = matchingConcept;
```
Comment (samuelmale) [inline src/components/encounter/ohri-encounter-form.component.tsx:266]:
```suggestion
          field.meta.concept = matchingConcept;
```

**Review by samuelmale — APPROVED** [review]

**Pull request merged**